string-edit.el

```diff
@@ -82,15 +82,24 @@ This saves you from needing to manually escape characters."
       (indent-region beg end))))
 
 (defun se/find-original ()
-  (if (derived-mode-p 'js2-mode 'js-mode)
-      (se/js-strings-at-point)
-    (se/string-at-point)))
-
-(defun se/guess-at-major-mode ()
-  (save-excursion
+  (cond
+   ((derived-mode-p 'js2-mode 'js-mode)
+    (se/js-strings-at-point))
+   ((eq major-mode 'emacs-lisp-mode)
+    (se/elisp-string-at-point))
+   (t
+    (se/string-at-point))))
```

Comment:
Can this process be applied to any language?

I have an idea.

```el
(defvar se/string-functions-alist
  (list '(js-mode . se/js-strings-at-point)
        '(js2-mode . se/js-strings-at-point)
        ;; '(js-jsx-mode . se/js-strings-at-point)
        ;; '(js2-jsx-mode . se/js-strings-at-point)
        ))

(defun se/find-original ()
  (if (assq major-mode se/string-functions-alist)
      (funcall (cdr (assq major-mode se/string-functions-alist)))
    (se/string-at-point)))
```

Reply:
How about using this instead of `assq`:
```emacs-lisp
(assoc major-mode se/string-functions-alist
       (lambda (parent child)
         (provided-mode-derived-p child parent)))
;;; or
(-some
 (lambda (cons)
   (and (derived-mode-p (car cons))
        (cdr cons)))
 se/string-functions-alist)
```